### PR TITLE
Add limits and requests for functions

### PR DESCRIPTION
**Pull request by alexellis**
Signed-off-by: Alex Ellis <alexellis2@gmail.com>

## Description

Add limits / requests for memory in line with the OpenFaaS Docker Swarm implementation and CLI

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

Results:
- [x] Limits parsed
- [x] Errors in parsing handled
- [ ] Limits set - still not working.
- [x] Behaviour for create
- [x] Behaviour for update
  
Pushed up via image:

```
        image: functions/faas-netesd:0.3.5-limits
```

Sample:

```
 curl -X POST 127.0.0.1:31111/system/functions --data-binary '{"image": "functions/alpine:latest", "service": "env", "envprocess": "env", "limits": {"memory": "100Mi"}, "network": "func_functions"}' -i
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [ ] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: handlers/deploy.go [hunk 1]
```diff
@@ -272,3 +282,29 @@ func createSelector(constraints []string) map[string]string {
...
+		Requests: apiv1.ResourceList{},
+	}
+
+	if request.Limits != nil && len(request.Limits.Memory) > 0 {
+		qty, err := resource.ParseQuantity(request.Limits.Memory)
+		if err != nil {
+			return resources, err
+		}
+
+		resources.Limits.Memory().Add(qty)
```
Comment (errm) [inline handlers/deploy.go]: try: `resources.Limits[apiv1.ResourceMemory] = qty`

Comment (alexellis) [reply handlers/deploy.go]: Thanks 👍

**Comment by ericstoekl** [comment]
Tested it out and it works. Confirmed that the `env` function has 100M limits with `kubectl describe pod` on the gateway:

```
    Limits:
      memory:  100Mi
    Requests:
      memory:  100Mi
```

 Then, confirmed that a normally deployed function does not have this `Limits:` and `Requests:` fields.

**Comment by alexellis** [comment]
Thank you @ericstoekl

**Comment by alexellis** [comment]
@resouer 👍 @LucasRoesler 👍

**Pull request merged**